react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -0,0 +1,266 @@
+      expressionResult = assetTrigger?.includes(condition.parameter) ?? undefined;
+      break;
+    }
+    case 'startsWith': {
+      expressionResult = assetTrigger?.startsWith(condition.parameter) ?? undefined;
+      break;
+    }
+    case 'endsWith': {
+      expressionResult = assetTrigger?.endsWith(condition.parameter) ?? undefined;
+      break;
+    }
+  }
+
+  return expressionResult;
+};
+const checkNumericExpressionStatement = (
+  asset: Asset,
+  expression: NumericExpression
+): boolean | undefined => {
+  if (!isMetadataTrigger(expression.trigger)) return undefined;
+
+  const trigger = expression.trigger;
+  const condition = expression.condition;
+
+  let expressionResult: boolean = false;
+
+  const assetTrigger = Number(asset[trigger.type]?.[trigger.key]);
+
+  switch (condition.type) {
+    case 'equals': {
+      const parameter = condition.parameters[0];
+      expressionResult = assetTrigger === parameter;
+      break;
+    }
+    case 'notEquals': {
+      const parameter = condition.parameters[0];
+      expressionResult = assetTrigger !== parameter;
+      break;
+    }
+    case 'lessThan': {
+      const parameter = condition.parameters[0];
+      expressionResult = assetTrigger < parameter;
+      break;
+    }
+    case 'greaterThan': {
+      const parameter = condition.parameters[0];
+      expressionResult = assetTrigger > parameter;
+      break;
+    }
+    case 'lessThanOrEquals': {
+      const parameter = condition.parameters[0];
+      expressionResult = assetTrigger <= parameter;
+      break;
+    }
+    case 'greaterThanOrEquals': {
+      const parameter = condition.parameters[0];
+      expressionResult = assetTrigger >= parameter;
+      break;
+    }
+    case 'within': {
+      const lower = condition.lowerBoundInclusive;
+      const upper = condition.upperBoundInclusive;
+      const value = assetTrigger;
+      expressionResult = lower < value && value < upper;
+      break;
+    }
+    case 'outside': {
+      const lower = condition.lowerBoundExclusive;
+      const upper = condition.upperBoundExclusive;
+      const value = assetTrigger;
+      expressionResult = value <= lower && upper <= value;
+      break;
+    }
+  }
+
+  return expressionResult;
+};
+
+const traverseExpression = (
+  asset: Asset,
+  expressions: Expression[]
+): Array<boolean | undefined> => {
+  let expressionResult: boolean | undefined = false;
+
+  const expressionResults: Array<boolean | undefined> = [];
+
+  expressions.forEach((expression) => {
+    switch (expression.type) {
+      case 'or': {
+        const operatorResult = traverseExpression(asset, expression.expressions);
+        expressionResult = operatorResult.find((result) => result) ?? false;
+        break;
+      }
+      case 'and': {
+        const operatorResult = traverseExpression(asset, expression.expressions);
+        expressionResult =
+          operatorResult.find((result) => (result !== undefined ? !result : false)) ?? false;
```

Comment:
Can this be just a `operatorResult.every((result) => result === true)` ?

Reply:
changed here
https://github.com/cognitedata/reveal/pull/4210/commits/03b40b181de31e95d5b302162f9a3d5d0b1ffb51